configure script fails to detect Erlang version on Alpine Linux aarch64 due to shell expands $-

Alpine Linux aarch64 builder fails to detect Erlang version:
````
...
checking for erl... /usr/bin/erl
checking for Erlang >= 19.3... configure: error: in `/home/buildozer/aports/community/cloudi/src/cloudi-1.7.4/src':
configure: error: test Erlang program execution failed
See `config.log' for more details
````

The config log reveals that the Erlang code is generated wrongly:
````
...
configure:23009: checking for Erlang >= 19.3
configure:23099: $ERLC $ERLCFLAGS -b beam conftest.$ac_ext >&5 && echo "#!/bin/s
h" > conftest$ac_exeext && $as_echo "\"$ERL\" -run conftest start -run init stop
 -noshell" >> conftest$ac_exeext && chmod +x conftest$ac_exeext
conftest.erl:16: syntax error before: 'end'
conftest.erl:2: function start/0 undefined
configure:23099: $? = 1
configure: program exited with status 1
configure: failed program was:
| -module(conftest).
| -export([start/0]).
| 
| start() ->
| 
|             Version = case erlang:system_info(otp_release) of
|                 [$R, Major1, Major2, T1 | MinorT] ->
|                     %% before 17.0
|                     Minor = if
|                         MinorT == [] ->
|                             0;
|                         true ->
|                             %% Remove -1 from R16B03-1
|                             list_to_integer(lists:takewhile(fun(C) ->
|                                 C /= 
|                             end, MinorT))
|                     end,
|                     if
|                         T1 == $B ->
|                             list_to_integer([Major1, Major2]) * 1000000 +
|                             Minor * 1000 + 999;
|                         T1 == $A ->
|                             list_to_integer([Major1, Major2]) * 1000000 +
|                             Minor * 1000
|                     end;
...
````

Note that line 16 it refers to has:
````erlang
                             list_to_integer(lists:takewhile(fun(C) ->
                                 C /= 
                             end, MinorT))
````
when it should be:
````erlang
                             list_to_integer(lists:takewhile(fun(C) ->
                                 C /= $-
                             end, MinorT))
````

This is because the shell [expands](http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_05_02) `$-` to the shell's current options flags. I believe this expansion is not intentional.

Fix is trivial:
````diff
--- a/src/m4/ax_erlang_otp_version.m4
+++ b/src/m4/ax_erlang_otp_version.m4
@@ -168,7 +168,7 @@ AC_DEFUN([AX_ERLANG_REQUIRE_OTP_VER],
                         true ->
                             %% Remove -1 from R16B03-1
                             list_to_integer(lists:takewhile(fun(C) ->
-                                C /= $-
+                                C /= \$-
                             end, MinorT))
                     end,
                     if
````

--- a/src/m4/ax_erlang_otp_version.m4
+++ b/src/m4/ax_erlang_otp_version.m4
@@ -168,7 +168,7 @@ AC_DEFUN([AX_ERLANG_REQUIRE_OTP_VER],
                         true ->
                             %% Remove -1 from R16B03-1
                             list_to_integer(lists:takewhile(fun(C) ->
-                                C /= $-
+                                C /= \$-
                             end, MinorT))
                     end,
                     if

